impeller/aiks/experimental_canvas.cc

```diff
@@ -371,13 +376,25 @@ void ExperimentalCanvas::SaveLayer(
       /*flood_input_coverage=*/!!backdrop_filter         //
   );
 
-  if (!maybe_subpass_coverage.has_value() ||
-      maybe_subpass_coverage->IsEmpty()) {
-    Save(total_content_depth);
-    return;
+  if (!maybe_subpass_coverage.has_value()) {
+    return SkipUntilMatchingRestore(total_content_depth);
   }
+
   auto subpass_coverage = maybe_subpass_coverage.value();
 
+  // When an image filter is present, clamp to avoid flicking due to nearest
+  // sampled image. For other cases, round out to ensure than any geometry is
+  // not cut off.
```

Comment:
This fixes https://github.com/flutter/flutter/issues/152366

Reply:
Yes, you will be filtering all of the subpass pixels, not just its theoretical coverage.